llvm/lib/Target/AIE/AIEBaseSubtarget.cpp

```diff
@@ -444,10 +444,35 @@ class WAWEdges : public ScheduleDAGMutation {
     auto *RI = static_cast<const AIEBaseRegisterInfo *>(TRI);
     LivePhysRegs LiveRegs;
     LiveRegs.init(*TRI);
-    // Reserved registers are considered always live
-    for (MCPhysReg PhysReg : MRI.getReservedRegs().set_bits()) {
-      if (RI->isSimplifiableReservedReg(PhysReg))
-        LiveRegs.addReg(PhysReg);
+    auto Scheduler = DAG->getBB()
+                         ? static_cast<AIEScheduleDAGMI *>(DAG)->getSchedImpl()
+                         : nullptr;
+    const BlockState *BS;
```

Comment:
Please avoid un-initialized pointers, they are very painful to debug :D Maybe we can make the pointer local to the `if (Scheduler)` scope?